### PR TITLE
Repro libcurl 401s

### PR DESCRIPTION
**Pull request by johnhany97**
Reproing https://github.com/mamba-org/mamba/issues/1132

Running this locally after setting up the dev environment will lead to this snippet:
```
127.0.0.1 - - [24/Aug/2021 15:01:29] "GET /noarch/repodata.json HTTP/1.1" 401 -
Got Basic KG5pbCk6dGVzdA==
Got Basic KG5pbCk6dGVzdA==
127.0.0.1 - - [24/Aug/2021 15:01:31] "GET /osx-64/repodata.json HTTP/1.1" 401 -
127.0.0.1 - - [24/Aug/2021 15:01:31] "GET /osx-64/repodata.json HTTP/1.1" 401 -
Got Basic KG5pbCk6dGVzdA==
127.0.0.1 - - [24/Aug/2021 15:01:31] "GET /osx-64/repodata.json HTTP/1.1" 401 -
Got Basic KG5pbCk6dGVzdA==
127.0.0.1 - - [24/Aug/2021 15:01:31] "GET /noarch/repodata.json HTTP/1.1" 401 -
Got Basic KG5pbCk6dGVzdA==
127.0.0.1 - - [24/Aug/2021 15:01:31] "GET /noarch/repodata.json HTTP/1.1" 401 -
Got
127.0.0.1 - - [24/Aug/2021 15:01:31] "GET /noarch/repodata.json HTTP/1.1" 401 -
Got Basic KG5pbCk6dGVzdA==
127.0.0.1 - - [24/Aug/2021 15:01:33] "GET /osx-64/repodata.json HTTP/1.1" 401 -
Got Basic KG5pbCk6dGVzdA==
127.0.0.1 - - [24/Aug/2021 15:01:33] "GET /osx-64/repodata.json HTTP/1.1" 401 -
Got Basic KG5pbCk6dGVzdA==
127.0.0.1 - - [24/Aug/2021 15:01:33] "GET /osx-64/repodata.json HTTP/1.1" 401 -
Got Basic KG5pbCk6dGVzdA==
127.0.0.1 - - [24/Aug/2021 15:01:33] "GET /noarch/repodata.json HTTP/1.1" 401 -
Got Basic KG5pbCk6dGVzdA==
127.0.0.1 - - [24/Aug/2021 15:01:33] "GET /noarch/repodata.json HTTP/1.1" 401 -
Got
127.0.0.1 - - [24/Aug/2021 15:01:33] "GET /noarch/repodata.json HTTP/1.1" 401 -
```

Noting how _some_ of the requests have nothing logged for the auth header, while the others do.

### CODE REVIEW

**Comment by wolfv** [comment]
Would you have a chance to rerun this with the latest `mamba` and `libcurl` (7.79.1)?

File: libmamba/src/core/fetch.cpp [hunk 1]
```diff
@@ -168,6 +205,11 @@ namespace mamba
         // - fixing how the progress bar works
         curl_easy_setopt(handle, CURLOPT_HTTP_VERSION, CURL_HTTP_VERSION_1_1);
 
+        if (std::getenv("MAMBA_SET_USEDPWD"))
+        {
+            curl_easy_setopt(handle, CURLOPT_USERPWD, ":test");
```
Comment (johnhany97) [inline libmamba/src/core/fetch.cpp:210]: @wolfv i'm pretty sure I rootcaused what was going on here.

I believe that with https://github.com/curl/curl/pull/6545/ (which was released with the release of libcurl where we noticed this starting to break for us), the fact that we never directly set the user/password on the curl handle but instead relied on it being passed through the URL led to some edge case in curl where the auth ends up being dropped from reusing the connection.

I'll follow up with an issue towards curl to see if we can get this sorted there.

Comment (johnhany97) [reply libmamba/src/core/fetch.cpp:210]: Filed https://github.com/curl/curl/issues/8449

**Comment by nicornk** [comment]
@johnhany97 Thank you for tracking this down and leading to a fix in curl. I was just running into this issue today.
Hope we can easily pull in the next curl release to fix this in mamba :)

**Comment by wolfv** [comment]
sooo, does this mean that the issue is solved? :tada:

Thanks a ton that you went to the great length of identifying a curl bug. Really really happy!

**Comment by johnhany97** [comment]
@wolfv yeah I think we can consider this done. I'll clean up the PR to keep the bump to libcurl. Would you also like me to keep the multi-server test given it was valuable in helping us repro this mode of failure?

**Comment by wolfv** [comment]
sounds good!

**Comment by johnhany97** [comment]
I've gone ahead and opened https://github.com/mamba-org/mamba/pull/1569 so that it's a bit cleaner. Going to close this one as it's served it's purpose :D

**Pull request closed without merging**